cms/djangoapps/contentstore/helpers.py

```diff
@@ -323,6 +323,58 @@ def import_staged_content_from_user_clipboard(parent_key: UsageKey, request) ->
     return new_xblock, notices
 
 
+def fetch_and_set_upstream_link(
+    copied_from_block: str | None,
+    copied_from_version_num: int,
+    temp_xblock: XBlock,
+    user: User
+):
+    """
+    Fetch and set upstream link for the given xblock. This function handles following cases:
+    * the xblock is copied from a v2 library; the library block is set as upstream.
+    * the xblock is copied from a course; no upstream is set, only copied_from_block is set.
+    * the xblock is copied from a course where the source block was imported from a library; the original libary block
+      is set as upstream.
+    """
+    if copied_from_block is None:
+        return
```

Comment:
[opinion] Can we make `copied_from_block` a required string parameter instead of returning here? I think that would be clearer.

```suggestion
    assert copied_from_block
```

Reply:
Did not add assert, but updated the type and moved the check to caller function.